pkg/ssh/pool.go

```diff
@@ -54,6 +56,33 @@ func startPool(ctx context.Context, serverAddr string, config *ssh.ClientConfig)
 	return p, nil
 }
 
+func retryNewClientConn(ctx context.Context, c net.Conn, addr string, conf *ssh.ClientConfig) (ssh.Conn, <-chan ssh.NewChannel, <-chan *ssh.Request, error) {
+	ticker := time.NewTicker(300 * time.Millisecond)
+	to := config.GetTimeout() / 6 // 5 seconds
+	timeout := time.Now().Add(to)
+
+	log.Infof("waiting for ssh to be ready")
+	for i := 0; ; i++ {
+		clientConn, chans, reqs, err := ssh.NewClientConn(c, addr, conf)
+		if err == nil {
+			return clientConn, chans, reqs, nil
+		}
+		log.Infof("ssh is not ready yet: %s", err)
+
+		if time.Now().After(timeout) {
+			return nil, nil, nil, err
+		}
+
+		select {
+		case <-ticker.C:
+			continue
+		case <-ctx.Done():
+			log.Infof("ssh.retryNewClientConn cancelled")
```

Comment:
done returns an err (or you can get it from the ctx), I find it's useful to log it to know the reason of the cancellation.

Reply:
ok